crates/matrix-sdk-common/src/executor.rs

```diff
@@ -64,6 +64,16 @@ impl<T> JoinHandle<T> {
     }
 }
 
+#[cfg(target_arch = "wasm32")]
+impl<T> Drop for JoinHandle<T> {
+    fn drop(&mut self) {
+        // don't abort the spawned future
```

Comment:
This comment must be elaborated to explain why we don't abort the future when the join handle is dropped:

- Why?
- Does it match the Tokio's behaviour?

Thanks.

Reply:
Yes, it matches tokio's behavior and that's exactly the reason. I think we discussed it on Matrix previously that a cancel-on-drop `JoinHandle` might make more sense for the SDK, but then you would have to
- wrap the tokio `JoinHandle` for non-wasm
- update all the existing cases where `JoinHandle`s are dropped prematurely to store the handle somewhere, or call some `detach` method on it

That's a lot harder. I think you should accept a PR that fixes the existing inconsistency (which leads to buggy behavior), and evaluate the different `JoinHandle` semantics afterwards.